File: 1350-remove-sub-folders-from-the-filesystem/1350-remove-sub-folders-from-the-filesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {
        sort(folder.begin(),folder.end(),[](string &a, string &b){
            return a.length()<b.length();
        });
        vector<string> res;
        map<string,int> ump;
        for(auto s: folder){
            string temp = "";
            bool found = false;
            for(int i = 0;i<s.length();i++){
                if(s[i]>='a' && s[i]<='z')  temp+=s[i];
                else{
                    if(ump.find(temp)!=ump.end()){
                        found = true;
                        break;
                    }
                    temp+=s[i];
                }
            }
            if(!found) {
                ump[temp]++;
                res.push_back(temp);
            }
        }
        return res;
    }
};

File: 1350-remove-sub-folders-from-the-filesystem/1350-remove-sub-folders-from-the-filesystem.py
class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        folder.sort(key = lambda x: len(x))
        seen = defaultdict(int)
        result = []
        for path in folder:
            temp = ""
            seenalready = False
            for char in path:
                if temp!="" and char=='/' and seen[temp] > 0:
                    seenalready = True
                temp+=char
            if not seenalready:
                result.append(path)
                seen[path]+=1

        return result 
